lib/puppet/file_bucket/file.rb

```diff
@@ -138,7 +137,7 @@ def checksum_data(base_method)
       Puppet::Util::Checksums.method(:"#{base_method}_file").call(@path)
     end
 
-    def to_s
+    def to_binary
       Puppet::FileSystem::binread(@path)
```

Comment:
One side effect of calling `binread` or `File.new(path, 'rb')` is that the encoding of the returned string will be `ASCII-8BIT`. But I don't know if any of our tests actually verify it. For example, if `binread` started returning a string in `UTF-8`, then ruby's net/http code might (not sure) corrupt the data when it serializes it.